mitodl/ocw-hugo-themes
# create lists of disallowed URLs for the base and www sitemaps

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-hugo-themes/issues/696

#### What's this PR do?
In https://github.com/mitodl/ocw-hugo-themes/pull/679, we added sitemap templates that render a sitemap at the root with a `sitemapindex` element that links in a sitemap for `ocw-www` (`sitemap-www.xml`) as well as all of the course sitemaps for known course sites as determined by an API call to `ocw-studio`.  It was noticed that inside the course and www sitemaps that there are some URLs that lead to blank pages.  These are caused by us not providing a list template for certain types of content, such as pages and resources as well as built in Hugo functionality we aren't using such as tags and categories.  This PR iterates a list of "disallowed urls" in each sitemap template and ignores pages matching those URLs when generating the sitemap.  In the future if we add templates for these types of content or utilize Hugo functionality like tags or categories, we can remove them from these lists and they will be included in the sitemap once again.

#### How should this be manually tested?
 - Clone `ocw-www` and at least one course repo from `mitocwcontent`
 - Set the following env variables:
```
WWW_CONTENT_PATH=/path/to/mitocwcontent/ocw-www/
COURSE_CONTENT_PATH=/path/to/mitocwcontent/
OCW_TEST_COURSE=<your test course>
```
 - Run `npm run start:www`
 - Visit http://localhost:3000/sitemap.xml and verify that the links contained in the sitemap go to valid pages
 - Run `npm run start:course`
 - Visit  http://localhost:3000/sitemap.xml and verify that the links contained in the sitemap go to valid pages
